.github/workflows/e2e-main.yml

```diff
@@ -148,11 +148,15 @@ jobs:
       env:
         TERM: xterm
 
-    - name: Upload Cypress recording upon failure
+    - name: Generate Mochawesome Reports
+      if: failure()
```

Comment:
I believe that `failure` in this context doesn't necessarily means Cypress failed. It could be that the job failed for literary any reason. Still, in more than 90-95% of the cases it will be Cypress.

Reply:
Just used the same logic as used here https://github.com/metabase/metabase/pull/23603/files#diff-42420760680849afeaba9ddbbc9c898a1bf90c5d03d7b9f8d8266513f614a5f2R155 where your statement also apply